spec/unit/provider/osx_profile_spec.rb

```diff
@@ -0,0 +1,259 @@
+          'PayloadVersion' => 1,
+          'PayloadIdentifier' => 'com.testprofile.screensaver',
+          'PayloadUUID' => '73fc30e0-1e57-0131-c32d-000c2944c108',
+          'PayloadEnabled' => true,
+          'PayloadDisplayName' => 'com.apple.screensaver',
+          'PayloadContent' => {
+            'com.apple.screensaver' => {
+              'Forced' => [
+                {
+                  'mcx_preference_settings' => {
+                    'idleTime' => 0,
+                  }
+                }
+              ]
+            }
+          }
+        }
+      ]
+      }
+      @no_profiles = {"_computerlevel"=>
+          [],
+      }
+      allow(@provider).to receive(:cookbook_file_available?).and_return(true)
+      allow(@provider).to receive(:cache_cookbook_profile).and_return('/tmp/test.mobileconfig.remote')
+      allow(@provider).to receive(:get_new_profile_hash).and_return(@test_profile)
+      allow(@provider).to receive(:get_installed_profiles).and_return(@all_profiles)
+      allow(@provider).to receive(:read_plist).and_return(@all_profiles)
+      allow(::File).to receive(:unlink).and_return(true)
+    end
+
+    it 'should build the get all profiles shellout command correctly' do
+      profile_name = 'com.testprofile.screensaver.mobileconfig'
+      tempfile = '/tmp/allprofiles.plist'
+      @new_resource.profile_name profile_name
+      allow(@provider).to receive(:generate_tempfile).and_return(tempfile)
+      allow(@provider).to receive(:get_installed_profiles).and_call_original
+      allow(@provider).to receive(:read_plist).and_return(@all_profiles)
+      expect(@provider).to receive(:shell_out!).with("profiles -P -o '/tmp/allprofiles.plist'")
+      @provider.load_current_resource
+    end
+
+    it 'should use profile name as profile when no profile is set' do
+      profile_name = 'com.testprofile.screensaver.mobileconfig'
+      @new_resource.profile_name profile_name
+      @provider.load_current_resource
+      expect(@new_resource.profile_name).to eql(profile_name)
+    end
+
+    it 'should use identifier from specified profile' do
+      @new_resource.profile @test_profile
+      @provider.load_current_resource
+      expect(
+        @provider.instance_variable_get(:@new_profile_identifier)
+        ).to eql(@test_profile['PayloadIdentifier'])
+    end
+
+    it 'should install when not installed' do
+      @new_resource.profile @test_profile
+      allow(@provider).to receive(:get_installed_profiles).and_return(@no_profiles)
+      @provider.load_current_resource
+      expect { @provider.run_action(:install) }.to_not raise_error
+    end
+
+    it 'does not install if the profile is already installed' do
+      @new_resource.profile @test_profile
+      allow(@provider).to receive(:get_installed_profiles).and_return(@all_profiles)
+      @provider.load_current_resource
+      expect(@provider.action_install).to_not receive(:install_profile)
```

Comment:
You're making sure that `@provider` does not receive `:install_profile`, correct?  If so, I don't think you have this expectation setup correctly.  I believe it needs to look like

```ruby
@provider.load_current_resource
expect(@provider).to_not receive(:install_profile)
expect { @provider.action_install }.to_not raise_error
```